<launch>
	<arg name="robot_namespace"  	default="drone0" />
	<arg name="robot_config_path" 	default="$(env AEROSTACK_STACK)/configs/$(arg robot_namespace)" />
	<arg name="rviz_config_path" 	default="$(find droneArchitectureRvizROSModule)/launch" />
	<arg name="frequency" 			default="200.0" />

	<!-- Config file -->
	<arg name="config_file" 		default="ekf_state_estimator_config.xml" />

	<!--Subscribers -->
	<arg name="estimated_speed" 	default="self_localization/speed" />
	<arg name="estimated_pose" 	    default="self_localization/pose" />
	<arg name="altitude" 	    	default="sensor_measurement/altitude" />
	<arg name="sensor_speed" 	    default="sensor_measurement/speed" />
	<arg name="battery_state" 	    default="sensor_measurement/battery_state" />
	<arg name="magnetometer" 	    default="sensor_measurement/magnetometer" />
	<arg name="pressure" 	        default="sensor_measurement/pressure" />
	<arg name="temperature" 	    default="sensor_measurement/temperature" />
	<arg name="roll_pitch" 	        default="actuator_command/roll_pitch" />
	<arg name="altitude_yaw_rate" 	default="actuator_command/altitude_rate_yaw_rate" />
	<arg name="flight_action" 	    default="actuator_command/flight_action" />
	<arg name="flight_state" 	    default="self_localization/flight_state" />

	<!-- Rviz -->
	<node pkg="rviz" type="rviz" name="rviz" args="-d $(arg rviz_config_path)/rviz_config.rviz"/>
	<node name="droneArchitectureRvizInterface" pkg="droneArchitectureRvizROSModule" type="droneArchitectureRvizInterfaceROSModuleNode" output="screen">	
  		<param name="drone_id_namespace" value="$(arg robot_namespace)" type="string" />
	</node>

	<group ns="$(arg robot_namespace)">
		<node name="quadrotor_simulator" pkg="quadrotor_simulator_process" type="quadrotor_simulator_process_main" output="screen">
			<param name="robot_namespace" 		value="$(arg robot_namespace)" 		type="string"/>
			<param name="frequency" 			value="$(arg frequency)" 			type="double"/>
			<param name="robot_config_path" 	value="$(arg robot_config_path)" 	type="string"/>
			<param name="config_file" 			value="$(arg config_file)" 			type="string"/> 
            <param name="estimated_speed"   	value="$(arg estimated_speed)"      type="string"/> 
            <param name="estimated_pose"    	value="$(arg estimated_pose)"       type="string"/>
			<param name="altitude"     			value="$(arg altitude)"       		type="string"/> 
			<param name="sensor_speed"     		value="$(arg sensor_speed)"       	type="string"/> 
			<param name="battery_state"     	value="$(arg battery_state)"       	type="string"/> 
			<param name="magnetometer"     		value="$(arg magnetometer)"       	type="string"/> 
			<param name="pressure"     			value="$(arg pressure)"       		type="string"/> 
			<param name="temperature"     		value="$(arg temperature)"       	type="string"/>
            <param name="roll_pitch"         	value="$(arg roll_pitch)"           type="string"/> 
            <param name="altitude_yaw_rate"  	value="$(arg altitude_yaw_rate)"    type="string"/>
            <param name="flight_action"      	value="$(arg flight_action)"        type="string"/> 
            <param name="flight_state"       	value="$(arg flight_state)"         type="string"/>
		</node>
		<node name="rviz_translator_from_simulator" pkg="droneSimulatorRvizROSModule" type="SimulatorRvizROSModuleNode" respawn="false" required="true" output="screen"> </node>
	</group>
</launch>